### DISCUSSION TITLE
When subscripting, if opc server disconnect can make error -- "Exception in thread Thread-2". I don't want change souce code but need to inherit many classes.

### DISCUSSION THREAD

**Discussion by CodePythonFollow** [FreeOpcUa/python-opcua#1191] [Q&A]
This is a bug report. to get help go to https://github.com/FreeOpcUa/python-opcua/discussions

**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior incl code:

**Expected behavior**

**Screenshots**

**Comment by swamper123** [comment]
Hi,

please provide the informations, which are mentioned. Don't write everything in the title.

**Comment by CodePythonFollow** [comment]
![image](https://user-images.githubusercontent.com/52147782/101738758-95d98a00-3b01-11eb-804f-e853be5d4119.png)

**Comment by CodePythonFollow** [comment]
![image](https://user-images.githubusercontent.com/52147782/101738844-b4d81c00-3b01-11eb-856d-1a866fd59d7d.png)
main thread can't get the error

**Comment by swamper123** [comment]
Catching Exceptions, caused by Threads, in Python is a pain (since ever). You can't pass them to the main thread.
You have to integrate signals or deal with the Exceptions in the Threads themselve or at least join the thread anywhere.

**Comment by CodePythonFollow** [comment]
Sure, thank you.Global variable as a integrate signals is good idea.

**Reply by swamper123** [reply]
I thought more about [signal](https://docs.python.org/3/library/signal.html?highlight=signal#module-signal).
Global variables can be a dangerous field. I hope you know, what you are doing.
